tmc-langs-r/src/test/java/fi/helsinki/cs/tmc/langs/r/RPluginTest.java

```diff
@@ -0,0 +1,203 @@
+    }
+
+    @After
+    public void tearDown() {
+        removeAvailablePointsJson(simpleAllTestsPassProject);
+        removeResultsJson(simpleAllTestsPassProject);
+        removeResultsJson(simpleSomeTestsFailProject);
+        removeResultsJson(simpleSourceCodeErrorProject);
+    }
+
+    private void removeResultsJson(Path projectPath) {
+        File resultsJson = new File(projectPath.toAbsolutePath().toString()
+                + "/.results.json");
+        resultsJson.delete();
+    }
+
+    private void removeAvailablePointsJson(Path projectPath) {
+        File availablePointsJson = new File(projectPath.toAbsolutePath().toString()
+                + "/.available_points.json");
+        availablePointsJson.delete();
+    }
+
+    private void testResultAsExpected(TestResult result, boolean successful, String name,
+                                      String[] points) {
+        assertEquals(successful, result.isSuccessful());
+        assertEquals(name, result.getName());
+        assertArrayEquals(points, result.points.toArray());
+    }
+
+    @Test
+    public void testGetTestCommand() {
+        String[] command = new String[]{"Rscript"};
+        String[] args;
+        if (SystemUtils.IS_OS_WINDOWS) {
+            args = new String[]{"-e", "\"library('tmcRtestrunner');run_tests()\""};
+        } else {
+            args = new String[]{"-e", "library(tmcRtestrunner);run_tests()"};
+        }
+        String[] expectedCommand = ArrayUtils.addAll(command, args);
+        Assert.assertArrayEquals(expectedCommand, plugin.getTestCommand());
+    }
+
+    @Test
+    public void testGetAvailablePointsCommand() {
+        String[] command = new String[]{"Rscript"};
+        String[] args;
+        if (SystemUtils.IS_OS_WINDOWS) {
+            args = new String[]{"-e", "\"library('tmcRtestrunner');run_available_points()\""};
+        } else {
+            args = new String[]{"-e", "library(tmcRtestrunner);run_available_points()"};
+        }
+        String[] expectedCommand = ArrayUtils.addAll(command, args);
+        Assert.assertArrayEquals(expectedCommand, plugin.getAvailablePointsCommand());
+    }
+
+    @Test
+    public void testGetPluginName() {
+        assertEquals("r", plugin.getLanguageName());
+    }
+
+    @Test
+    public void testScanExercise() {
+        plugin.scanExercise(simpleAllTestsPassProject, "main.R");
+        File availablePointsJson = new File(simpleAllTestsPassProject.toAbsolutePath().toString()
+                + "/.available_points.json");
+
+        assertTrue(availablePointsJson.exists());
+    }
+
+    @Test
+    public void testScanExerciseInTheWrongPlace() {
+        plugin.scanExercise(simpleAllTestsPassProject, "ar.R");
+        Path availablePointsJson = simpleAllTestsPassProject.resolve(".available_points.json");
+        ImmutableList<TestDesc> re = null;
+        try {
+            re = new RExerciseDescParser(availablePointsJson).parse();
+        } catch (IOException e) {
```

Comment:
Maybe just throw this exception from the test. It would certainly make this method a bit cleaner.